src/jit/lower.cpp

```diff
@@ -960,7 +960,16 @@ GenTreePtr Lowering::NewPutArg(GenTreeCall* call, GenTreePtr arg, fgArgTabEntryP
 #endif // FEATURE_MULTIREG_ARGS
 #endif // not defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
         {
-            putArg = comp->gtNewOperNode(GT_PUTARG_REG, type, arg);
+#ifdef _TARGET_ARM_
+            if (comp->opts.compUseSoftFP && type == TYP_DOUBLE)
+            {
+                putArg = new (comp, GT_PUTARG_REG) GenTreeMultiRegOp(type, arg, nullptr);
```

Comment:
Would this also be required for passing `TYP_LONG`? Also, I think you need to be consistent in using `GenTreeMultiRegOp` for `GT_PUTARG_REG`, or you should use a different oper (e.g. `GT_PUTARG_MULTIREG`), since the JIT (mostly) depends on the oper being a "tag" for the `GenTree` type. That is, a given oper should always be on the same `GenTree` type.